daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/PackageInterface.scala

```diff
@@ -293,47 +293,38 @@ private[lf] class PackageInterface(signatures: PartialFunction[PackageId, Packag
       }
     )
 
-  import PackageInterface.ChoiceInfo
-
-  private[lf] def lookupChoice(
-      identifier: TypeConName,
+  // look
+  private[lf] def lookupLenientChoice(
+      templateId: TypeConName,
       chName: ChoiceName,
-  ): Either[LookupError, ChoiceInfo] = {
-    lazy val context = Reference.Choice(identifier, chName)
-    lookupTemplateOrInterface(identifier, context).flatMap {
-      case Left(template) =>
-        template.choices.get(chName) match {
-          case Some(choice) => Right(ChoiceInfo.Template(choice))
-          case None =>
-            template.inheritedChoices.get(chName) match {
-              case Some(ifaceId) =>
-                lookupInterfaceChoice(ifaceId, chName, context).map(
-                  ChoiceInfo.Inherited(ifaceId, _)
-                )
-              case None =>
-                Left(LookupError(context, context))
-            }
-        }
-      case Right(interface) =>
-        interface.fixedChoices.get(chName) match {
-          case Some(choice) => Right(ChoiceInfo.Interface(choice))
-          case None => {
-            // TODO(drsk) improve the performance of this lookup. Tracked in issue
-            // https://github.com/digital-asset/daml/issues/13630.
-            interface.requires.view
-              .map((iface) =>
-                lookupInterfaceChoice(iface, chName, context).map((choice) => (choice, iface))
-              )
-              .collectFirst({ case Right((choice, iface)) => (choice, iface) }) match {
-              case Some((choice, iface)) =>
-                Right(ChoiceInfo.InterfaceInherited(iface, choice))
-              case None => Left(LookupError(context, context))
-            }
+  ): Either[LookupError, PackageInterface.ChoiceInfo] = {
+    lazy val context = Reference.Choice(templateId, chName)
+    lookupTemplate(templateId, context).flatMap { template =>
+      template.choices.get(chName) match {
+        case Some(choice) =>
+          Right(PackageInterface.ChoiceInfo.Template(choice))
+        case None =>
+          template.inheritedChoices.get(chName) match {
```

Comment:
What happens with a collision here?

Reply:
The collision check is still in place, so nothing problematic happens.
If we disable the check, we will just get one random choice among the one that collides.

This is currenlty only use to provide a backward compatible API for Canton until the update their protocol (to track of the `interfaceId` field of the ExerciseNode). I am not sure it is worthwhile to make it more sensible. 

We could decide to make the ledger API more lenient: try to infer the choice among all the inherited choice  and crash if it is ambiguous. We could reuse this lookup for this purpose. In this case we will change it to produces error in case of collision. 

I will add a comment to remove it  (together with `ReplayCommand.LenientExercise`) once Canton have fix their protocol.